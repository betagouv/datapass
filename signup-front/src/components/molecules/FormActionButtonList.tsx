import { FunctionComponent, MouseEvent } from 'react';
import {
  EnrollmentAction,
  userInteractionsConfiguration,
} from '../../lib/enrollment-actions-configuration';
import FormActionButton from '../atoms/FormActionButton';
import ButtonGroup from './ButtonGroup';

export const listAuthorizedActions = (acl: Record<string, boolean>) =>
  (Object.keys(userInteractionsConfiguration) as EnrollmentAction[]).filter(
    (key) => acl[key]
  );

type Props = {
  enrollment: any;
  loading: boolean;
  pendingAction?: EnrollmentAction;
  onActionButtonClick: Function;
};

const FormActionButtonList: FunctionComponent<Props> = ({
  enrollment,
  loading,
  pendingAction,
  onActionButtonClick,
}) => {
  const authorizedActions = listAuthorizedActions(enrollment.acl);

  return (
    <ButtonGroup alignRight>
      {authorizedActions.map((action) => {
        const actionConfiguration = userInteractionsConfiguration[action];
        const onClick = (event: MouseEvent<HTMLElement>) => {
          event.preventDefault();
          onActionButtonClick(action);
        };
        const isPendingAction = action === pendingAction;
        return (
          <FormActionButton
            key={action}
            loading={loading}
            onClick={onClick}
            isPendingAction={isPendingAction}
            {...actionConfiguration.displayProps}
          />
        );
      })}
    </ButtonGroup>
  );
};

export default FormActionButtonList;
