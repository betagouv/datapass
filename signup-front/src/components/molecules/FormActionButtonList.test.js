import { listAuthorizedActions } from './FormActionButtonList';

describe('FormActionButtonList', () => {
  it('lists button to display from enrollment acl', () => {
    const acl = {
      create: false,
      delete: true,
      destroy: false,
      edit: true,
      index: false,
      new: false,
      notify: true,
      refuse_application: true,
      review_application: false,
      send_application: true,
      show: true,
      update: true,
      update_owner: false,
      update_rgpd_contact: false,
      validate_application: false,
    };

    expect(listAuthorizedActions(acl)).toMatchSnapshot();
  });
});
