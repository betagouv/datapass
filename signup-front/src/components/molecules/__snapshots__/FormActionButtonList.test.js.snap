// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormActionButtonList lists button to display from enrollment acl 1`] = `
Array [
  "notify",
  "update",
  "send_application",
  "refuse_application",
]
`;
